### PR TITLE
* : manual cherry-pick 8126 to release 6.5

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #8115 

#### Tests 

 - Manual test (add detailed scripts or steps below)
1. create changefeed, found `sourceID=1` in cdc.log
2. pause changefeed
3. execute `set global tidb_source_id=6;` in the upstream TiDB.
4. resume changefeed, found `sourceID=6` in cdc.log

### Release note 

```release-note
Fix the problem causing changefeed creation to fail due to inability to retrieve the sourceID from PD.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,14 +40,23 @@ ifeq (${CDC_ENABLE_VENDOR}, 1)
...
+ifeq (${OS}, "linux")
+	CGO := 0
+else ifeq (${OS}, "darwin")
+	CGO := 1
+endif
+
+GOBUILD  := CGO_ENABLED=$(CGO) $(GO) build $(BUILD_FLAG) -trimpath $(GOVENDORFLAG)
 GOBUILDNOVENDOR  := CGO_ENABLED=0 $(GO) build $(BUILD_FLAG) -trimpath
-GOTEST   := CGO_ENABLED=1 $(GO) test -p $(P) --race
+GOTEST   := CGO_ENABLED=1 $(GO) test -p $(P) --race --tags=intest
```
Comment (lance6716) [inline Makefile:57]: this is not needed? I remember the change at TiDB side is not picked to release-6.5

File: dm/tests/print_status/run.sh [hunk 2]
```diff
@@ -74,7 +74,7 @@ function check_print_status() {
 
 	# check load unit print status
 	status_file=$WORK_DIR/worker1/log/loader_status.log
-	grep -oP "\[unit=lightning-load\] \[IsCanceled=false\] \[finished_bytes=59637\] \[total_bytes=59637\] \[progress=.*\]" $WORK_DIR/worker1/log/dm-worker.log >$status_file
+	grep -oP "\[unit=lightning-load\] \[IsCanceled=false\] \[finished_bytes=59834\] \[total_bytes=59834\] \[progress=.*\]" $WORK_DIR/worker1/log/dm-worker.log >$status_file
```
Comment (lance6716) [inline dm/tests/print_status/run.sh:77]: not sure if the value should be updated

File: engine/test/integration_tests/dm_collation/run.sh [hunk 3]
```diff
@@ -39,7 +39,7 @@ function run() {
 	# wait for dump and load finished
 
 	exec_with_retry --count 30 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.task_status.\"mysql-01\".status.unit == \"DMSyncTask\" and .task_status.\"mysql-02\".status.unit == \"DMSyncTask\"'"
-	curl http://127.0.0.1:10245/api/v1/jobs/$job_id/status | tee /dev/stderr | jq -e '.finished_unit_status."mysql-02"[1].Status.finishedBytes == 174'
+	curl http://127.0.0.1:10245/api/v1/jobs/$job_id/status | tee /dev/stderr | jq -e '.finished_unit_status."mysql-02"[1].Status.finishedBytes == 248'
```
Comment (lance6716) [inline engine/test/integration_tests/dm_collation/run.sh:42]: and here

**Review by lance6716 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (asddongmen) [reply Makefile:57]: Yes, we need it, otherwise the unit test will fail.

**Pull request merged**